python/simulators/CMakeLists.txt

```diff
@@ -20,6 +20,13 @@ add_custom_command(
     DEPENDS ${PYTHON_DOCSTRINGS_FILE}
     COMMENT "Generating PyBlackOilSimulatorDoc.hpp from JSON file"
 )
+
+find_file(DOCSTRINGS_JSON docstrings_common.json
+  PATHS ${opm-common_DIR} ${opm-common_PYTHON_COMMON_DIR}
+  PATH_SUFFIXES python NO_DEFAULT_PATH REQUIRED)
+
+configure_file(${DOCSTRINGS_JSON} ${PROJECT_SOURCE_DIR}/python/docstrings_common.json COPYONLY)
```

Comment:
I guess this copy to the project source directory (instead of to the binary directory) is not ideal. And since the github action runs separately from the cmake build, this file would not appear there anyway. It looks like you added a GitHub action in https://github.com/OPM/opm-common/pull/4018 to transfer the file directly instead? It is an interesting idea, but it looks a little bit dangerous :) Maybe the GitHub action here in opm-simulators could instead checkout the docstring json file from opm-common and pass that to the sphinx build scripts?

Reply:
Yes, the copy command would only be neccessary for building the documentation locally.

I've added a GitHub action to transfer the file directly from opm-common to here so a new build is triggered if that file in opm-common is changed. If I add a github action here, then a new build of the documentation would not be triggered if that file in opm-common is changed, or is that possible somehow?

Reply:
Yes good point, I did not think of that we also need to trigger the action in opm-simulators immediately such that the docs there are in sync with the change of the file in opm-common... But then, instead of having the workflow in opm-common copy the file to opm-simulators to trigger the sphinx build, can we instead have it trigger the opm-simulators workflow directly? (Which will copy the file from opm-common and build the docs). See https://stackoverflow.com/q/70151645/2173773 for an example.

Reply:
I've removed the copy command for now.

Reply:
I've added the docstrings_common.json to the .gitignore file now, now the github action here copies the file from the opm-common repository and the documentation is rebuilt also when the file in opm-common changes.
For local builds, this file is be copied from opm-common to here.